app/controllers/api/public/v1/application_controller.rb

```diff
@@ -10,7 +10,7 @@ class ApplicationController < PublicAPIController
 
         def recruitment_cycle
           year = params[:recruitment_cycle_year]
-          @recruitment_cycle ||= RecruitmentCycle.find_by(year:) || RecruitmentCycle.current_recruitment_cycle
+          @recruitment_cycle ||= RecruitmentCycle.find_by(year:) || RecruitmentCycle.current_recruitment_cycle!
```

Comment:
A suggestion here (at your discretion) is to put this behind an environment check so that this would only return a 404 and not raise a sentry error in non prod environments because the recruitment cycle _should_ always be found in prod and we'd want to know if it wasn't there.